tests/trestle/core/commands/validate_test.py

```diff
@@ -280,7 +288,116 @@ def test_validate_catalog_params(sample_catalog_rich_controls: Catalog) -> None:
     """Test validation of unique param ids in catalog."""
     args = argparse.Namespace(mode=const.VAL_MODE_CATALOG)
     validator: Validator = validator_factory.get(args)
-    assert validator.model_is_valid(sample_catalog_rich_controls, True)
+    assert validator.model_is_valid(sample_catalog_rich_controls, True, None)
     param_0_id = sample_catalog_rich_controls.groups[0].controls[0].params[0].id
     sample_catalog_rich_controls.groups[0].controls[0].params[1].id = param_0_id
-    assert not validator.model_is_valid(sample_catalog_rich_controls, False)
+    assert not validator.model_is_valid(sample_catalog_rich_controls, False, None)
+
+
+def test_rule_param_values_validator_happy(tmp_trestle_dir: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
+    """Test validation of rules in SSP."""
+    args = argparse.Namespace(mode=const.VAL_MODE_RULES)
+    validator: Validator = validator_factory.get(args)
+    prof_name = 'comp_prof'
+    ssp_name = 'new_ssp'
+
+    gen_args, _ = setup_for_ssp(tmp_trestle_dir, prof_name, ssp_name)
+    args_compdefs = gen_args.compdefs
+
+    # first create the markdown
+    ssp_gen = SSPGenerate()
+    assert ssp_gen._run(gen_args) == 0
+
+    # first ssp assembly
+    ssp_assemble = f'trestle author ssp-assemble -m {ssp_name} -o {ssp_name} -cd {args_compdefs}'
+    test_utils.execute_command_and_assert(ssp_assemble, 0, monkeypatch)
+
+    new_ssp, _ = ModelUtils.load_model_for_class(tmp_trestle_dir, ssp_name, ossp.SystemSecurityPlan)
```

Comment:
This code block repeats 3 times in the tests, please create a helper function to not duplicate the code.

Reply:
Done